include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm.hpp

```diff
@@ -215,39 +227,47 @@ class pairwise_alignment_algorithm : protected policies_t...
      * phase all remaining cells are computed. In the final phase the last cell is possibly evaluated for a new
      * alignment optimum.
      */
-    template <std::ranges::input_range alignment_matrix_column_t, std::ranges::input_range sequence2_t>
-    void initialise_column(alignment_matrix_column_t && first_column, sequence2_t && sequence2)
+    template <std::ranges::input_range alignment_column_t,
+              std::ranges::input_range cell_index_column_t,
+              std::ranges::input_range sequence2_t>
+    void initialise_column(alignment_column_t && alignment_column,
+                           cell_index_column_t && cell_index_column,
+                           sequence2_t && sequence2)
     {
         // ---------------------------------------------------------------------
         // Initial phase: prepare column and initialise first cell
         // ---------------------------------------------------------------------
 
-        auto first_column_iter = first_column.begin();
-        *first_column_iter = this->track_cell(this->initialise_origin_cell());
+        auto first_column_it = alignment_column.begin();
+        auto cell_index_column_it = cell_index_column.begin();
+        *first_column_it = this->track_cell(this->initialise_origin_cell(), *cell_index_column_it);
 
         // ---------------------------------------------------------------------
         // Iteration phase: iterate over column and compute each cell
         // ---------------------------------------------------------------------
 
         for ([[maybe_unused]] auto && unused : sequence2)
         {
-            ++first_column_iter;
-            *first_column_iter = this->track_cell(this->initialise_first_column_cell(*first_column_iter));
+            ++first_column_it;
+            *first_column_it = this->track_cell(this->initialise_first_column_cell(*first_column_it),
+                                                *++cell_index_column_it);
```

Comment:
I remember from your reviews of my sam/bam code that you said dereferencing a temporary iterator is not recommended (or even UB?)?

Reply:
Yes it is. But that‘s not a temporary. Pre increment always returns a reference to the iterator.